src/TypeInference/PolyExpr.ml

```diff
@@ -39,7 +39,7 @@ let method_call_ctx ~pos ~env ~self ~self_tp ~self_r_eff ~eff =
     (result_expr, res_tp, ret_effect_join self_r_eff r_eff)
   | _ ->
     (* Method must be an arrow with monomorphic argument *)
-    InterpLib.InternalError.report ~reason:"invalid method type" ()
+    Error.fatal (Error.invalid_method_type ~pos)
```

Comment:
Is it error or internal error? Do you have any example input that causes this error?

Reply:
This change might have been done without proper understanding